### PR TITLE
Add multiple particles source

### PR DESCRIPTION
**Pull request by dpgrote**
This adds a new particle source, allowing specification of multiple particles (in contrast to the single particle source). I've found this handy in some of my tests and it's a small enough change.

Note that there is no CI test yet included. I can easily modify one of the tests using the single particle source to use the multiple particle source instead (but still load one particle). Alternatively, there are several tests that load multiple particles using multiple species (the photon_pusher and radiation_reaction tests). One of these could be restructured to use the multiple particle source, but that would be a fair amount of work since the analysis script would also have to be updated to expect a single species instead of the multiple species. It would be easier if the authors of the test could make this change. Comments welcome.

### CODE REVIEW

File: Docs/source/running_cpp/parameters.rst [hunk 1]
```diff
@@ -400,6 +400,16 @@ Particle initialization
       ``<species_name>.single_particle_vel`` (`3 doubles`, particle 3D normalized momentum, i.e. :math:`\gamma \beta`)
       ``<species_name>.single_particle_weight`` ( `double`, macroparticle weight, i.e. number of physical particles it represents)
 
+    * ``MultipleParticles``: Inject multiple macroparticles.
```
Comment (lucafedeli88) [inline Docs/source/running_cpp/parameters.rst:403]: Just one minor comment: here the injection styles are called `SingleParticle` and `MultipleParticles`, however in `PlasmaInjector.cpp` we write `part_pos_s == "singleparticle"` and  `part_pos_s == "multipleparticles"`. I guess that the parser is **case-insensitive** for these directives (although it is **case-sensitive** for user-defined constants), but this is never mentioned in `https://warpx.readthedocs.io/en/latest/running_cpp/parameters.html`. Maybe we should add somewhere in that page a paragraph to explain what is case-insensitive and what is case-sensitive...

**Review by lucafedeli88 — COMMENTED** [review]
Thanks for this PR! I think that this is a very useful feature... I only left a minor comment

File: Docs/source/running_cpp/parameters.rst (hunk 1 above)
Comment (dpgrote) [reply Docs/source/running_cpp/parameters.rst:403]: @lucafedeli88 You raise a good point. That should be done is a separate PR. As part of that a check should be done to make sure that in fact WarpX is consistent in how it handles case sensitivity.

**Comment by ax3l** [comment]
Restarting CI to make sure coverage is on

File: Docs/source/running_cpp/parameters.rst (hunk 1 above)
Comment (lucafedeli88) [reply Docs/source/running_cpp/parameters.rst:403]: I agree with you

**Comment by ax3l** [comment]
@dpgrote can you please rebase against  the latest `development` (or alternatively merge the latest `development`)? We fixed some CI timeout issues yesterday.

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**